### ISSUE TITLE
downloader.go:302 sync busy

### ISSUE THREAD

**Issue by splix** [ethereumproject/go-ethereum#184]
too many:
```
I0413 09:01:18.983868 downloader.go:302] sync busy
```

it seems to be it needs some improvement, or a bug somewhere

**Comment by whilei** [comment]
I was wondering about this too -- had assumed it was just my slow HDD.

**Comment by whilei** [comment]
Just seeing this and wondering if a slow/poorly-connected peer might be part of the problem (just judging by output correlation; not decisive, but perhaps suggestive?): 
```
~/Desktop/geth_trials  $ geth --fast --datadir=gethdata --chainconfig=cc_11.json
I0417 16:47:16.516470 flag.go:326] INFO: Global --datadir flag is set: gethdata
I0417 16:47:16.582328 cmd/geth/flag.go:594] Using custom chain configuration file: cc_11.json
I0417 16:47:16.582438 cmd/geth/flag.go:152] Found custom bootstrap nodes in config file: [enode://e809c4a2fec7daed400e5e28564e23693b23b2cc5a019b612505631bbe7b9ccf709c1796d2a3d29ef2b045f210caf51e3c4f5b6d3587d43ad5d6397526fa6179@174.112.32.157:30303 enode://6e538e7c1280f0a31ff08b382db5302480f775480b8e68f8febca0ceff81e4b19153c6f8bf60313b93bef2cc34d34e1df41317de0ce613a201d1660a788a03e2@52.206.67.235:30303 enode://5fbfb426fbb46f8b8c1bd3dd140f5b511da558cd37d60844b525909ab82e13a25ee722293c829e52cb65c2305b1637fa9a2ea4d6634a224d5f400bfe244ac0de@162.243.55.45:30303 enode://42d8f29d1db5f4b2947cd5c3d76c6d0d3697e6b9b3430c3d41e46b4bb77655433aeedc25d4b4ea9d8214b6a43008ba67199374a9b53633301bca0cd20c6928ab@104.155.176.151:30303 enode://814920f1ec9510aa9ea1c8f79d8b6e6a462045f09caa2ae4055b0f34f7416fca6facd3dd45f1cf1673c0209e0503f02776b8ff94020e98b6679a0dc561b4eba0@104.154.136.117:30303 enode://72e445f4e89c0f476d404bc40478b0df83a5b500d2d2e850e08eb1af0cd464ab86db6160d0fde64bd77d5f0d33507ae19035671b3c74fec126d6e28787669740@104.198.71.200:30303 enode://5cd218959f8263bc3721d7789070806b0adff1a0ed3f95ec886fb469f9362c7507e3b32b256550b9a7964a23a938e8d42d45a0c34b332bfebc54b29081e83b93@35.187.57.94:30303 enode://39abab9d2a41f53298c0c9dc6bbca57b0840c3ba9dccf42aa27316addc1b7e56ade32a0a9f7f52d6c5db4fe74d8824bcedfeaecf1a4e533cacb71cf8100a9442@144.76.238.49:30303 enode://f50e675a34f471af2438b921914b5f06499c7438f3146f6b8936f1faeb50b8a91d0d0c24fb05a66f05865cd58c24da3e664d0def806172ddd0d4c5bdbf37747e@144.76.238.49:30306]
I0417 16:47:16.583113 ethdb/database.go:63] Alloted 128MB cache and 1024 file handles to gethdata/chaindata
I0417 16:47:16.691720 cmd/geth/flag.go:594] Using custom chain configuration file: cc_11.json
I0417 16:47:16.691811 cmd/geth/flag.go:570] Found custom chain configuration: &{[%!s(*core.Fork=&{Homestead true false 0xc42295a500 <nil> [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0xc420010740}) %!s(*core.Fork=&{ETF false true 0xc42295a720 <nil> [148 54 94 58 140 11 53 8 156 29 17 149 8 31 231 72 155 82 138 132 178 33 153 201 22 24 13 184 178 138 222 127] <nil>}) %!s(*core.Fork=&{GasReprice true false 0xc42295a780 <nil> [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0xc420010d40}) %!s(*core.Fork=&{Diehard true false 0xc42295a9e0 0xc42295aa20 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0xc420010e40})] [%!s(*core.BadHash=&{0xc42295acc0 [5 190 243 14 245 114 39 15 101 71 70 218 34 99 154 122 12 151 221 151 167 5 11 158 37 35 145 153 106 174 182 137]})] 61}
I0417 16:47:16.692356 cmd/geth/flag.go:279] WARNING: No etherbase set and no accounts found as default
I0417 16:47:16.756686 cmd/geth/flag.go:594] Using custom chain configuration file: cc_11.json
I0417 16:47:16.756737 ethdb/database.go:63] Alloted 128MB cache and 1024 file handles to gethdata/chaindata
I0417 16:47:17.198276 flag.go:484] successfully wrote genesis block and allocations: d4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3
I0417 16:47:17.199779 ethdb/database.go:63] Alloted 128MB cache and 1024 file handles to gethdata/chaindata
I0417 16:47:17.392104 eth/backend.go:595] upgrading db log bloom bins
I0417 16:47:17.392575 eth/backend.go:603] upgrade completed in 479.434µs
I0417 16:47:17.392622 ethdb/database.go:63] Alloted 16MB cache and 16 file handles to gethdata/dapp
I0417 16:47:17.395881 eth/backend.go:154] Protocol Versions: [63 62], Network Id: 1, Chain Id: 61
I0417 16:47:17.395999 eth/backend.go:183] Blockchain DB Version: 3
I0417 16:47:17.396747 core/blockchain.go:206] Last header: #0 [d4e56740…] TD=17179869184
I0417 16:47:17.396764 core/blockchain.go:207] Last block: #0 [d4e56740…] TD=17179869184
I0417 16:47:17.396775 core/blockchain.go:208] Fast block: #0 [d4e56740…] TD=17179869184
I0417 16:47:17.397202 p2p/server.go:310] Starting Server
I0417 16:47:17.400850 p2p/nat/nat.go:111] mapped network port udp:30303 -> 30303 (ethereum discovery) using NAT-PMP(10.0.1.1)
I0417 16:47:17.403798 p2p/discover/udp.go:215] Listening, enode://26ca693e53198a14ecba0da36e4b50b644d03f61e537894bd4cc6e51bf98550fc5f1afb916e3a1dbc313602e5e7ecc8e19884bf80afa1a8074db27dc1a41b821@75.132.105.176:30303
I0417 16:47:17.407198 p2p/server.go:553] Listening on [::]:30303
I0417 16:47:17.407784 node/node.go:295] IPC endpoint opened: gethdata/geth.ipc
I0417 16:47:17.417093 p2p/nat/nat.go:111] mapped network port tcp:30303 -> 30303 (ethereum p2p) using NAT-PMP(10.0.1.1)
I0417 16:47:53.578997 eth/downloader/downloader.go:335] Block synchronisation started
I0417 16:47:53.941656 downloader.go:302] sync busy
I0417 16:47:53.995916 downloader.go:302] sync busy
I0417 16:48:00.355072 downloader.go:302] sync busy
I0417 16:48:00.803998 downloader.go:302] sync busy
I0417 16:48:01.733984 downloader.go:302] sync busy
I0417 16:48:02.114418 downloader.go:302] sync busy
I0417 16:48:08.602723 downloader.go:302] sync busy
I0417 16:48:09.053554 downloader.go:302] sync busy
I0417 16:48:09.080240 downloader.go:302] sync busy
I0417 16:48:09.513374 downloader.go:302] sync busy
I0417 16:48:15.699365 downloader.go:302] sync busy
I0417 16:48:17.720403 downloader.go:302] sync busy
I0417 16:48:20.856766 downloader.go:302] sync busy
I0417 16:48:22.006769 downloader.go:302] sync busy
I0417 16:48:24.848030 downloader.go:302] sync busy
I0417 16:48:25.858637 downloader.go:302] sync busy
I0417 16:48:27.682446 downloader.go:302] sync busy
I0417 16:48:28.364906 downloader.go:302] sync busy
I0417 16:48:28.881190 downloader.go:302] sync busy
I0417 16:48:29.292299 downloader.go:302] sync busy
I0417 16:48:29.871285 downloader.go:302] sync busy
I0417 16:48:30.441031 downloader.go:302] sync busy
I0417 16:48:36.249390 downloader.go:302] sync busy
I0417 16:48:36.288005 downloader.go:302] sync busy
I0417 16:48:39.705321 downloader.go:302] sync busy
I0417 16:48:40.650341 downloader.go:302] sync busy
I0417 16:48:41.003682 downloader.go:302] sync busy
I0417 16:48:48.943337 downloader.go:302] sync busy
I0417 16:48:53.584201 downloader.go:307] peer "7ce45ecdb106c91b" drop: timeout
I0417 16:48:55.095319 downloader.go:302] sync busy
I0417 16:48:57.900885 core/headerchain.go:299] imported 384 header(s) (0 ignored) in 1.385547334s. #384 [88e96d45… / d3d5d5c1…]
I0417 16:48:57.917823 core/blockchain.go:743] imported 2 receipt(s) (0 ignored) in 346.101µs. #2 [88e96d45… / b495a1d7…]
I0417 16:48:59.840194 core/headerchain.go:299] imported 1152 header(s) (0 ignored) in 199.473225ms. #1536 [bbb506ab… / b787b8e0…]
I0417 16:49:03.206165 core/blockchain.go:743] imported 4 receipt(s) (0 ignored) in 663.064µs. #6 [3d612266… / 1f1aed8e…]
I0417 16:49:05.099592 downloader.go:302] sync busy
I0417 16:49:08.344342 core/blockchain.go:743] imported 1530 receipt(s) (0 ignored) in 576.474284ms. #1536 [e0c7c0b4… / b787b8e0…]
I0417 16:49:08.565154 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 773.87429ms. #3584 [25e5354f… / 46bbf165…]
I0417 16:49:08.748150 core/blockchain.go:743] imported 5 receipt(s) (0 ignored) in 2.525002ms. #1541 [25e5354f… / 90afd6b7…]
I0417 16:49:09.227198 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 597.841865ms. #5632 [2eb282d3… / 5dcf54ca…]
I0417 16:49:09.577546 core/blockchain.go:743] imported 666 receipt(s) (0 ignored) in 237.037013ms. #2207 [50eb45d4… / e4e4ce21…]
I0417 16:49:10.102456 core/blockchain.go:743] imported 1379 receipt(s) (0 ignored) in 459.376631ms. #3586 [99d0fd70… / 035c2c1e…]
I0417 16:49:10.320829 core/blockchain.go:743] imported 770 receipt(s) (0 ignored) in 208.148174ms. #4356 [c7f9d5d9… / d4a00afb…]
I0417 16:49:10.439150 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 1.116819144s. #7680 [55b73002… / ecdfeeeb…]
I0417 16:49:11.151482 core/blockchain.go:743] imported 1277 receipt(s) (0 ignored) in 462.785062ms. #5633 [1e440592… / 55b73002…]
I0417 16:49:11.365814 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 803.630109ms. #9728 [5db42d55… / b797851d…]
I0417 16:49:12.062897 core/blockchain.go:743] imported 2048 receipt(s) (0 ignored) in 437.665146ms. #7681 [345660e8… / 5db42d55…]
I0417 16:49:12.072005 core/blockchain.go:743] imported 16 receipt(s) (0 ignored) in 8.548406ms. #7697 [bd4cac07… / f3eac862…]
I0417 16:49:12.165084 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 727.099643ms. #11776 [c8ac595c… / f749a561…]
I0417 16:49:12.703559 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 475.605071ms. #13824 [2acef9df… / aba8811c…]
I0417 16:49:13.255945 core/blockchain.go:743] imported 1369 receipt(s) (0 ignored) in 437.462252ms. #9066 [7ad920d3… / af4713db…]
I0417 16:49:13.355857 eth/downloader/downloader.go:979] imported 1 state entries in 859.009µs: processed 1, pending at least 17
I0417 16:49:13.449252 core/blockchain.go:743] imported 683 receipt(s) (0 ignored) in 183.461234ms. #9749 [00ba5a0b… / b84db394…]
I0417 16:49:13.648054 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 864.643725ms. #15872 [32232e5b… / 4457b931…]
I0417 16:49:14.159132 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 451.955057ms. #17920 [94cf5e76… / 924938a3…]
I0417 16:49:14.290922 eth/downloader/downloader.go:979] imported 2 state entries in 392.762µs: processed 3, pending at least 49
I0417 16:49:14.650399 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 402.282743ms. #19968 [fa04ece3… / 585f7b44…]
I0417 16:49:14.926212 eth/downloader/downloader.go:979] imported 2 state entries in 4.362855ms: processed 5, pending at least 81
I0417 16:49:15.142798 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 420.647543ms. #22016 [a3da9eb0… / dbaffedf…]
I0417 16:49:15.321979 eth/downloader/downloader.go:979] imported 2 state entries in 941.389µs: processed 7, pending at least 113
I0417 16:49:15.502417 eth/downloader/downloader.go:979] imported 3 state entries in 1.749036ms: processed 10, pending at least 161
I0417 16:49:15.592609 eth/downloader/downloader.go:979] imported 6 state entries in 5.211732ms: processed 16, pending at least 257
I0417 16:49:15.745070 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 534.752806ms. #24064 [353cb2a8… / 256e05b8…]
I0417 16:49:15.796775 eth/downloader/downloader.go:979] imported 2 state entries in 9.579866ms: processed 18, pending at least 289
I0417 16:49:15.931971 core/headerchain.go:299] imported 512 header(s) (0 ignored) in 118.102046ms. #24576 [bf83be0d… / 32aa8325…]
I0417 16:49:16.176963 core/headerchain.go:299] imported 768 header(s) (0 ignored) in 197.053531ms. #25344 [5c99751f… / be208bf1…]
I0417 16:49:16.326565 eth/downloader/downloader.go:979] imported 2 state entries in 996.229µs: processed 20, pending at least 321
I0417 16:49:16.562265 core/headerchain.go:299] imported 1344 header(s) (0 ignored) in 331.185508ms. #26688 [3ddf4b9e… / 486883fc…]
I0417 16:49:16.601866 eth/downloader/downloader.go:979] imported 11 state entries in 19.596225ms: processed 31, pending at least 497
I0417 16:49:16.636958 core/headerchain.go:299] imported 192 header(s) (0 ignored) in 56.040593ms. #26880 [272e63e8… / ee37c07a…]
I0417 16:49:16.653894 eth/downloader/downloader.go:979] imported 2 state entries in 5.853146ms: processed 33, pending at least 529
I0417 16:49:16.756731 eth/downloader/downloader.go:979] imported 13 state entries in 2.202541ms: processed 46, pending at least 737
I0417 16:49:17.462593 eth/downloader/downloader.go:979] imported 4 state entries in 895.876µs: processed 50, pending at least 801
I0417 16:49:18.104335 eth/downloader/downloader.go:979] imported 21 state entries in 25.229843ms: processed 71, pending at least 1137
I0417 16:49:18.352943 eth/downloader/downloader.go:979] imported 24 state entries in 13.600511ms: processed 95, pending at least 1521
I0417 16:49:18.469565 core/blockchain.go:743] imported 2048 receipt(s) (0 ignored) in 663.282332ms. #11797 [063f93c3… / 13eb48da…]
I0417 16:49:18.912693 eth/downloader/downloader.go:979] imported 37 state entries in 17.330785ms: processed 132, pending at least 2113
I0417 16:49:18.966327 eth/downloader/downloader.go:979] imported 10 state entries in 2.746095ms: processed 142, pending at least 2273
I0417 16:49:19.101786 core/blockchain.go:743] imported 1903 receipt(s) (0 ignored) in 625.526713ms. #13700 [56729901… / 14ac8a60…]
I0417 16:49:19.347751 eth/downloader/downloader.go:979] imported 2 state entries in 16.053405ms: processed 144, pending at least 2305
I0417 16:49:19.434424 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 1.267282729s. #28928 [e955c3d3… / d0ff65a0…]
I0417 16:49:19.888459 eth/downloader/downloader.go:979] imported 93 state entries in 53.036107ms: processed 237, pending at least 3793
I0417 16:49:19.890457 core/blockchain.go:743] imported 2048 receipt(s) (0 ignored) in 743.286211ms. #15748 [eccc35a8… / 91ad7aea…]
I0417 16:49:20.412435 core/blockchain.go:743] imported 1781 receipt(s) (0 ignored) in 516.335977ms. #17529 [e28ce1d7… / 16a1574d…]
I0417 16:49:20.547326 core/blockchain.go:743] imported 415 receipt(s) (0 ignored) in 129.839201ms. #17944 [50949031… / 0029be0f…]
I0417 16:49:20.592991 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 1.018967967s. #30976 [7ba9a6cb… / aeaab5ad…]
I0417 16:49:20.760664 eth/downloader/downloader.go:979] imported 41 state entries in 6.967483ms: processed 278, pending at least 4449
I0417 16:49:21.078550 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 417.601753ms. #33024 [a57fb262… / bf1b31c6…]
I0417 16:49:21.231048 eth/downloader/downloader.go:979] imported 1 state entries in 570.157µs: processed 279, pending at least 4465
I0417 16:49:21.408907 core/headerchain.go:299] imported 960 header(s) (0 ignored) in 272.136214ms. #33984 [908760a3… / 2b73654f…]
I0417 16:49:21.931704 eth/downloader/downloader.go:979] imported 213 state entries in 239.691856ms: processed 492, pending at least 7873
I0417 16:49:21.941410 eth/downloader/downloader.go:979] imported 18 state entries in 247.599981ms: processed 510, pending at least 8161
I0417 16:49:22.030168 core/headerchain.go:299] imported 960 header(s) (0 ignored) in 583.059452ms. #34944 [a0a3c63f… / 91dca83a…]
I0417 16:49:22.562290 core/headerchain.go:299] imported 2048 header(s) (0 ignored) in 450.583241ms. #36992 [82392081… / b85f367b…]
I0417 16:49:23.028868 eth/downloader/downloader.go:979] imported 59 state entries in 35.521956ms: processed 569, pending at least 9105
I0417 16:49:23.085508 core/headerchain.go:299] imported 1216 header(s) (0 ignored) in 457.493097ms. #38208 [e4fa410c… / 5e082824…]
I0417 16:49:23.353041 core/blockchain.go:743] imported 2048 receipt(s) (0 ignored) in 579.065664ms. #19992 [141db0ea… / 86c1f2d7…]
I0417 16:49:23.705125 eth/downloader/downloader.go:979] imported 341 state entries in 153.538594ms: processed 910, pending at least 14561
I0417 16:49:23.814704 core/blockchain.go:743] imported 2048 receipt(s) (0 ignored) in 454.92943ms. #22040 [537236f2… / 2a6ba907…]
^CI0417 16:49:24.022842 cmd/geth/cmd.go:68] Got interrupt, shutting down...
I0417 16:49:24.023021 node/node.go:327] IPC endpoint closed: gethdata/geth.ipc
I0417 16:49:24.023388 core/blockchain.go:568] Chain manager stopped
I0417 16:49:24.023400 eth/handler.go:219] Stopping ethereum protocol handler...
I0417 16:49:24.062539 eth/handler.go:240] Ethereum protocol handler stopped
I0417 16:49:24.062571 core/tx_pool.go:165] Transaction pool stopped
I0417 16:49:24.062587 eth/backend.go:474] Automatic pregeneration of ethash DAG OFF (ethash dir: /Users/ia/.ethash)
F0417 16:49:24.062617 core/database_util.go:250] failed to store last header's hash into database: leveldb: closed
```

**Comment by whilei** [comment]
Closing in lieu of merged PR #202 _Problem: the name "sync busy" is confusing, is it too busy to sync?_
